From 507386b5774df4e76d24b189e3c8c3c79334086c Mon Sep 17 00:00:00 2001
From: Lars-Peter Clausen <lars@metafoo.de>
Date: Wed, 15 Jun 2016 14:20:16 +0200
Subject: [PATCH 28/30] ASoC: snd_soc_of_parse_daifmt(): Handle
 SND_SOC_DAIFMT_SPDIF

Allow to specify that a link use SPDIF format by setting the format to
"spdif" in the devicetree.

Signed-off-by: Lars-Peter Clausen <lars@metafoo.de>
---
 Documentation/devicetree/bindings/sound/simple-card.txt | 2 +-
 sound/soc/soc-core.c                                    | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/sound/simple-card.txt b/Documentation/devicetree/bindings/sound/simple-card.txt
index a4c72d09cd45..946a6db7cba9 100644
--- a/Documentation/devicetree/bindings/sound/simple-card.txt
+++ b/Documentation/devicetree/bindings/sound/simple-card.txt
@@ -49,7 +49,7 @@ Optional dai-link subnode properties:
 
 - format				: CPU/CODEC common audio format.
 					  "i2s", "right_j", "left_j" , "dsp_a"
-					  "dsp_b", "ac97", "pdm", "msb", "lsb"
+					  "dsp_b", "ac97", "pdm", "msb", "lsb", "spdif"
 - frame-master				: Indicates dai-link frame master.
 					  phandle to a cpu or codec subnode.
 - bitclock-master			: Indicates dai-link bit clock master.
diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index 473eefe8658e..a012788f9584 100644
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -3543,6 +3543,7 @@ unsigned int snd_soc_of_parse_daifmt(struct device_node *np,
 		{ "pdm",	SND_SOC_DAIFMT_PDM},
 		{ "msb",	SND_SOC_DAIFMT_MSB },
 		{ "lsb",	SND_SOC_DAIFMT_LSB },
+		{ "spdif",	SND_SOC_DAIFMT_SPDIF },
 	};
 
 	if (!prefix)
-- 
2.17.1

